### ISSUE TITLE
Cursor does not focus on file created when creating file with nested directory

### ISSUE THREAD

**Issue by GZLiew** [nvim-tree/nvim-tree.lua#1716]
### Description

Cursor does not focus on file created when creating file with nested directory

### Neovim version

```text
NVIM v0.7.2
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
```

### Operating system and version

Linux-5.19.12

### nvim-tree version

6ca6f99e7689c68679e8f0a58b421545ff52931f

### Minimal config

```lua
require("nvim-tree").setup({
  update_focused_file = {
    enable = true
  },
  git = {
    ignore = false
  },
  actions = {
    open_file = {
      quit_on_open = true
    }
  },
  view = {
    width = 200,
    number = true,
    relativenumber = true,
    side = 'left',
    mappings = {
      list = {
        { key = "-", cb = tree_cb("close") },
        { key = { "<BS>", "<C-o>" }, action = "close_node" },
      }
    }
  }
})
```

### Steps to reproduce

1. nvim
2. a
3. somefolder/anotherfolder/a.c
4. cursor focuses on the top of the tree

### Expected behavior

Cursor should focus on the top of the tree

![image](https://user-images.githubusercontent.com/43195293/199410881-34e7f52d-e3b3-4593-b8dd-689c285ea7c8.png)

### Actual behavior

Cursor should focus on the file created

![image](https://user-images.githubusercontent.com/43195293/199410960-84978be0-d47e-431b-8045-983f07d5e969.png)

**Comment by alex-courtis** [comment]
Confirmed:

Occurs whenever a file is created in a directory specified via the input.

Open / closed state of the directories has no effect.

**Comment by alex-courtis** [comment]
Problem: the current (focused) node is synchronously refreshed, but not the node in which the file is created.

Solutions:
1. open the tree to the folder containing the new file and focus it
2. focus the file only if the folder is already open

1 would be best achieved via a find file action
2 would require refreshing the folder if it is open

**Comment by alex-courtis** [comment]
Unrelated bug: `FileCreated` and `FolderCreated` events are dispatched regardless of whether a directory or file is created.

**Comment by alex-courtis** [comment]
@GZLiew I will be grateful if you can test the fix https://github.com/nvim-tree/nvim-tree.lua/pull/1722

```sh
cd /path/to/nvim-tree.lua
git pull
git checkout 1716-focus-file-dir-on-creation
```

**Comment by GZLiew** [comment]
hi @alex-courtis the fix works! Thanks for fixing! 

Tested with both single nest and multi nested directory

![image](https://user-images.githubusercontent.com/43195293/200107480-c0aec1c9-f5e0-4956-b5be-2a8683be5229.png)

**Comment by alex-courtis** [comment]
Unfortunately I have had to back out this change as it breaks `update_focused_file.enabled` and `renderer.group_empty`

Attempting fix again.

**Comment by alex-courtis** [comment]
Requires #1723

**Comment by alex-courtis** [comment]
Requires #1731

**Comment by alex-courtis** [comment]
Should be resolved by #1757

Please let me know if there are any issues.
